rust/lance-index/src/scalar/inverted/query.rs

```diff
@@ -0,0 +1,380 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use std::collections::HashSet;
+
+use lance_core::{Error, Result};
+use snafu::location;
+
+#[derive(Debug, Clone)]
+pub struct FtsSearchParams {
+    pub limit: usize,
+    pub wand_factor: f32,
+}
+
+impl FtsSearchParams {
+    pub fn new() -> Self {
+        Self {
+            limit: usize::MAX,
+            wand_factor: 1.0,
+        }
+    }
+
+    pub fn with_limit(mut self, limit: usize) -> Self {
+        self.limit = limit;
+        self
+    }
+
+    pub fn with_wand_factor(mut self, factor: f32) -> Self {
+        self.wand_factor = factor;
+        self
+    }
+}
+
+impl Default for FtsSearchParams {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+pub trait FtsQueryNode {
+    fn fields(&self) -> HashSet<String>;
```

Comment:
Minor nit:

Maybe slightly efficient to do `fn fields(&self, gathered: &mut HashSet<String>)`